pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -447,6 +467,9 @@ public RebalanceResult rebalance(TableConfig tableConfig, RebalanceConfig rebala
 
       // Check version and update IdealState
       try {
+        if (_tableRebalanceObserver.isAborted()) {
+          throw new RuntimeException(String.format("Rebalance for table: %s has been aborted", tableNameWithType));
```

Comment:
Will this cause `RebalanceResult` to be `UNKNOWN_ERROR`? We should make it explicit that it is cancelled

Reply:
this leads to FAILED status, as I put the abort checks inside try-catch blocks in TableRebalancer. the UNKNOWN_ERROR only happens when there is unexpected (uncaught) exceptions, and such case is tracked by an error counter to set alert on if needed.

Reply:
This is regular flow. Can we make it returning CANCELLED explicitly?

Reply:
added, so that metrics can track aborted/cancelled cases separately